### PR TITLE
updated genome_configs/*yml files

### PR DESCRIPTION
**Pull request by vpbrendel**
All *.yml files have been updated to at least have the current checksums so that fidibus should run without the --relax option.

For Aaeg the problem was that the download includes the mitochondrial genome and the corresponding annotation in the GFF3 file lacks mRNA entries for every gene, whereas CDS entries are present. The new yml file excludes the mitochondrial genome, which we want to do anyways.

For 28 species at least one of the download files was updated (mostly the GFF3 annotation). This suggests we should review all the exceptions/filters we have used; maybe some of them are no longer necessary. Help please - it's a lot of data checking.

For Nvit, the reference genome has changed to a newer long-read assembly. This would be a good example for using fidibus to quantify and visualize the status of an assembly - the new assembly has pseudo-chromosomes instead of a large number of scaffolds.

### CODE REVIEW

**Comment by timlai4** [comment]
Should we implement the feature that failed checksums merely flags a warning instead of stops the entire process?

**Comment by vpbrendel** [comment]
I thought that's exactly what the option --relax does.

**Comment by timlai4** [comment]
Sorry, I meant to make --relax the default and implement something like a --strict.

**Comment by vpbrendel** [comment]
Today's updates allow delta=0 (no adjustment to the input GFF3 file) and fix remaining accounting problems with the efficient length. The practical test for the latter is the following: efficient length of all iLoci up to an iiLocus should add up to the length of the sequence up to this point. The submitted code seems to pass all tests in that respect, independent of delta;

I don't know yet what the unit test failure is about.

**Comment by vpbrendel** [comment]
The test fails on

locuspocus --delta 0 amel-aug-nvit-param.gff3 

where the input file has a gene starting at position 1.

**Comment by vpbrendel** [comment]
Ok, the last two commits should fix the edge conditions so that no fiLoci of zero length are created.

**Comment by vpbrendel** [comment]
I am not sure that the testing failures are legitimate. For example,

grape-pred.gff3:chr8	CpGAT	gene	10538	11678	.	-	.	ID=chr8.g2;Name=chr8.g2

but AgnLocusStream.c does not have entry for a start at 10538.
Daniel - can you please review?

**Comment by vpbrendel** [comment]
It looks like the old code run like this

locuspocus --delta=0 grape-refr.gff3 grape-pred.gff3

produces output identical to the new binary invoked with the option -y, as in

locuspocus --delta=0 -y grape-refr.gff3 grape-pred.gff3

This is a consequence of allowing the usual reporting of ii lengths also for delta=0. Thus, it seems to me that we should change this unit test to include the -y flag. I am sure Daniel will know how to do this ...

File: src/core/AgnLocusStream.c [hunk 1]
```diff
@@ -364,9 +365,12 @@ static void locus_stream_extend(AgnLocusStream *stream, AgnLocus *locus)
 
       if(stream->endmode <= 0 && !stream->skip_iiLoci)
       {
-        AgnLocus *iilocus = agn_locus_new(seqid);
         GtRange irange = { prev_range.end + stream->delta + 1,
                            locusrange.start - stream->delta - 1 };
+	//If the two iLoci overlap, do not create an iiLocus:
+        if(irange.start > irange.end) goto noiilocus;
+
```
Comment (standage) [inline src/core/AgnLocusStream.c]: If I'm reading things clearly, this is when handling internal loci. This will prevent the inference of ziLoci, will it not? The details aren't fresh in my mind, but some of the genome organization statistics depended on tracking zero-length iiLoci. Maybe I'm missing something, but it looks like this would affect that tracking.

File: src/core/AgnLocusStream.c [hunk 2]
```diff
@@ -261,7 +261,8 @@ static void locus_stream_extend(AgnLocusStream *stream, AgnLocus *locus)
     {
       agn_locus_set_range(locus, locusrange.start - stream->delta,
                           locusrange.end);
-      if(stream->endmode >= 0 && !stream->skip_iiLoci)
+      if(stream->endmode >= 0 && !stream->skip_iiLoci && 
+				locusrange.start - stream->delta - 1 > 0)
```
Comment (standage) [inline src/core/AgnLocusStream.c]: Looks like many of the changes to this file are cosmetic (line width). This is one of the few changes related to length accounting, right?

File: src/core/AgnLocusStream.c [hunk 3]
```diff
@@ -412,7 +417,8 @@ static void locus_stream_extend(AgnLocusStream *stream, AgnLocus *locus)
     {
       agn_locus_set_range(locus, locusrange.start,
                           locusrange.end + stream->delta);
-      if(stream->endmode >= 0 && !stream->skip_iiLoci)
+      if(stream->endmode >= 0 && !stream->skip_iiLoci &&
+			locusrange.end + stream->delta + 1 <= seqrange.end)
```
Comment (standage) [inline src/core/AgnLocusStream.c]: Looks like many of the changes to this file are cosmetic (line width). This is one of the few changes related to length accounting, right?

File: src/core/AgnLocusStream.c (hunk 1 above)
Comment (standage) [reply src/core/AgnLocusStream.c]: I found this searching in the GitHub repo's history for ziLocus: https://github.com/BrendelGroup/AEGeAn/commit/b2f4d2924c1568a4ee6685c9bba5c9704c48ea2c.

Comment (vpbrendel) [reply src/core/AgnLocusStream.c]: The changes to the accounting had the following in mind:
1) If two intervals overlap, then split the overlap evenly for determination of effective length.
2) The sum of effective lengths should add up to the overall sequence lengths in any bounded region (be the entire sequence, the entire genome, or any region bounded by iiLoci).
3) A user should be able to select delta=0 just as well as delta=1, delta=100, or delta=500.

These principles were extensively tested on the Scer genome (which is compact).

Creation of ziLoci occurs elsewhere in the code, I believe, and we need to check whether it is necessary and done correctly.

Comment (standage) [reply src/core/AgnLocusStream.c]: Fair enough.

**Comment by standage** [comment]
I've tested out this branch on a Linux system (CentOS 7) and I still cannot reproduce the LocusPocus unit test failures. I just created a new branch "brendelupdate" with a minor fix for a failing functional test (since I can't push to vpbrendel/master, which wouldn't be a good idea even if I could). But I'm having a hard time troubleshooting the failure since I can't reproduce it.

**Comment by standage** [comment]
I'm seeing weird behavior. I created a new branch from `vpbrendel:master`, added some changes to fix the functional test failing on my end, created a new branch (`brendelupdate`), and then committed and pushed these to GitHub. The [tests pass just fine](https://travis-ci.org/BrendelGroup/AEGeAn/jobs/654406951?utm_medium=notification&utm_source=github_status) for this branch, but the diff between this branch and master only include my changes, not all of the changes in `vpbrendel:master`. I'll see if I can submit a pull request to add these changes to `vpbrendel:master`.

**Comment by vpbrendel** [comment]
Let's make sure that we are working on the same versions:

git clone https://github.com/vpbrendel/AEGeAn vbAEGeAn
cd vbAEGeAn
make all
make test

reliably produces the failed unit test anywhere I have looked. The failure seems to be based on the -y flag not being default after my changes, as discussed above.

**Comment by standage** [comment]
Something has definitely gotten jumbled somewhere in my setup. I can reproduce the bug now on a clean clone. Troubleshooting there...

**Comment by standage** [comment]
Ok, still not sure why things weren't lining up correctly. But I was able to fix the unit test in branch `fixunit`. @vpbrendel, if you can merge that branch into your master, I'm hoping that will resolve all the testing issues.
